### PR TITLE
Add badges to registry

### PR DESCRIPTION
**Pull request by grod220**
Closes: https://github.com/prax-wallet/registry/issues/97
Related: https://github.com/penumbra-zone/web/issues/1864

Adds the badge for the osmosis chain icon to all alloyed assets coming from Osmosis.

### CODE REVIEW

File: input/chains/penumbra-testnet-deimos-8.json
Comment (grod220) [inline input/chains/penumbra-testnet-deimos-8.json:1]: This chain input had an overlapping id with another input json. This means it did not render. Can delete.

File: npm/src/remote.test.ts
Comment (grod220) [inline npm/src/remote.test.ts:1]: New api on the fetch-mock library w/ version bump

File: tools/compiler/src/parser.rs [hunk 1]
```diff
@@ -31,41 +31,14 @@ pub struct ChainConfig {
...
-}
-
-pub trait IntoPbImages {
-    fn into_pb_images(self) -> Vec<AssetImage>;
-}
-
-impl IntoPbImages for Vec<Image> {
-    fn into_pb_images(self) -> Vec<AssetImage> {
-        self.into_iter().map(AssetImage::from).collect()
-    }
```
Comment (grod220) [inline tools/compiler/src/parser.rs:68]: Found defining a local image type to be unnecessary when we can just import the protobuf type directly

File: input/chains/penumbra-1.json
Comment (grod220) [inline input/chains/penumbra-1.json:1]: Labels all the "native" alloy assets coming from osmosis with a chain badge

File: input/chains/penumbra-testnet-deimos-8.json
Comment (grod220) [inline input/chains/penumbra-testnet-deimos-8.json:1]: This chain input had an overlapping id with another input json. This means it did not render. Can delete.

File: npm/src/remote.test.ts
Comment (grod220) [inline npm/src/remote.test.ts:1]: New api on the fetch-mock library w/ version bump

File: input/chains/penumbra-1.json
Comment (grod220) [inline input/chains/penumbra-1.json:1]: Labels all the "native" alloy assets coming from osmosis with a chain badge

File: npm/src/registry.test.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ describe('Registry', () => {
     const usdcId = base64ToUint8Array('reum7wQmk/owgvGMWMZn/6RFPV24zIKq3W6In/WwZgg=');
     const res = registry.getMetadata(new AssetId({ inner: usdcId }));
     expect(res.base).toEqual('wtest_usd');
+    expect(res.badges.length).toEqual(1);
+    expect(res.badges[0]?.png).toEqual(
+      'https://raw.githubusercontent.com/prax-wallet/registry/main/images/penumbra-favicon.png',
+    );
```
Comment (TalDerei) [inline npm/src/registry.test.ts:16]: test coverage can be expanded – assets without badges or missing metadata.

**Review by TalDerei — APPROVED** [review]

File: npm/src/registry.test.ts (hunk 2 above)
Comment (grod220) [reply npm/src/registry.test.ts:16]: Missing metadata test already there. Added test for getting asset without badge.

File: input/chains/penumbra-1.json [hunk 3]
```diff
@@ -128,5 +130,93 @@
...
     "transfer/channel-4/factory/osmo104jtrwcljnxfljhml8mxrw7qetcsdmqvy3sprw/ucosmousd": 10000,
     "transfer/channel-4/factory/osmo1g8qypve6l95xmhgc0fddaecerffymsl7kn9muw/squosmo": 10000
+  },
+  "badgesByBase": {
+    "transfer/channel-4/factory/osmo1em6xs47hd82806f5cxgyufguxrrc7l0aqx7nzzptjuqgswczk8csavdxek/alloyed/allUSDT": [
+      {
+        "png": "https://raw.githubusercontent.com/cosmos/chain-registry/master/osmosis/images/osmo.png",
+        "svg": "https://raw.githubusercontent.com/cosmos/chain-registry/master/osmosis/images/osmo.svg"
+      }
+    ],
```
[the remark this answers is not in the document]
Comment (grod220) [reply input/chains/penumbra-1.json:140]: Good suggestion, added

File: tools/compiler/src/processor.rs [hunk 4]
```diff
@@ -182,6 +183,18 @@ fn process_chain_config(chain_config: ChainConfig) -> AppResult<Registry> {
         }
     }
 
+    // add badges if available
+    for metadata in &mut all_metadata {
+        if let Some(badges) = chain_config
+            .badges_by_base
+            .get(&metadata.base_denom().denom)
+        {
+            let mut pb_metadata: pb::Metadata = metadata.clone().into();
```
[the remark this answers is not in the document]
Comment (grod220) [reply tools/compiler/src/processor.rs:192]: Can't do an into() from a mutable metadata domain type

**Pull request merged**